File: tcp_push_client.py
import socket

# Prompt the user for the server address and port
SERVER_HOST = input("Enter the host to send data to: ")
SERVER_PORT = int(input("Enter the port to send data to: "))

# Create a TCP/IP socket
client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

# Connect the socket to the server's address and port
try:
    client_socket.connect((SERVER_HOST, SERVER_PORT))
    print(f"Connected to {SERVER_HOST} on port {SERVER_PORT}")
    
    while True:
        # Prompt the user to enter the data to send
        data = input("Enter the message to send (or 'exit' to quit): ")
        
        # Exit the loop if the user types 'exit'
        if data.lower() == 'exit':
            print("Exiting the program.")
            break
        
        # Send the entered data to the server
        client_socket.sendall(data.encode('utf-8'))
        print(f"Sent data: {data}")

except Exception as e:
    print(f"Error: {e}")

finally:
    # Close the socket
    client_socket.close()
    print("Connection closed.")

File: tcp_push_client.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>

int main() {
    char SERVER_HOST[100];
    int SERVER_PORT;
    int client_socket;
    struct sockaddr_in server_addr;
    char data[1024];

    // Prompt the user for the server address and port
    printf("Enter the host to send data to: ");
    fgets(SERVER_HOST, sizeof(SERVER_HOST), stdin);
    SERVER_HOST[strcspn(SERVER_HOST, "\n")] = 0; // Remove newline character

    printf("Enter the port to send data to: ");
    scanf("%d", &SERVER_PORT);
    getchar(); // To consume the newline left by scanf

    // Create a TCP/IP socket
    client_socket = socket(AF_INET, SOCK_STREAM, 0);
    if (client_socket == -1) {
        perror("Socket creation failed");
        exit(1);
    }

    // Prepare server address structure
    server_addr.sin_family = AF_INET;
    server_addr.sin_port = htons(SERVER_PORT);
    server_addr.sin_addr.s_addr = inet_addr(SERVER_HOST);

    // Connect the socket to the server's address and port
    if (connect(client_socket, (struct sockaddr *)&server_addr, sizeof(server_addr)) == -1) {
        perror("Connection failed");
        close(client_socket);
        exit(1);
    }
    printf("Connected to %s on port %d\n", SERVER_HOST, SERVER_PORT);

    // Communicate with the server
    while (1) {
        // Prompt the user to enter the data to send
        printf("Enter the message to send (or 'exit' to quit): ");
        fgets(data, sizeof(data), stdin);
        data[strcspn(data, "\n")] = 0; // Remove newline character from input

        // Exit the loop if the user types 'exit'
        if (strcasecmp(data, "exit") == 0) {
            printf("Exiting the program.\n");
            break;
        }

        // Send the entered data to the server
        if (send(client_socket, data, strlen(data), 0) == -1) {
            perror("Send failed");
            break;
        }
        printf("Sent data: %s\n", data);
    }

    // Close the socket
    close(client_socket);
    printf("Connection closed.\n");

    return 0;
}
